pipeline/bicleaner/bicleaner.sh

```diff
@@ -15,36 +15,35 @@ corpus_prefix=$1
 output_prefix=$2
 bicleaner_threshold=$3
 type=$4
+threads=$5
+pack_dir=$6
 
 output_dir=$(dirname "${output_prefix}")
-tmp_dir="${output_dir}/tmp"
-mkdir -p "${tmp_dir}"
+mkdir -p "${output_dir}"
 
 if [ "${type}" == 'bicleaner-ai' ]; then
   echo "### Using bicleaner-ai"
-  bash "pipeline/bicleaner/download-pack.sh" "${tmp_dir}" "bicleaner-ai"
   cmd=bicleaner-ai-classify
 elif [ "${type}" == 'bicleaner' ]; then
   echo "### Using bicleaner"
-  bash "pipeline/bicleaner/download-pack.sh" "${tmp_dir}" "bicleaner"
   cmd=bicleaner-classify
 else
   echo "### Unsupported type: ${type}"
   exit 1
 fi
 
 echo "### Classifying and filtering"
-test -s "${tmp_dir}/best.gz" ||
+test -s "${output_prefix}.best.gz" ||
   paste <(pigz -dc "${corpus_prefix}.${SRC}.gz") <(pigz -dc "${corpus_prefix}.${TRG}.gz") |
-  ${cmd} --scol 1 --tcol 1 - - "${tmp_dir}"/*.yaml |
+  ${cmd} --scol 1 --tcol 1 --processes "${threads}"  - - "${pack_dir}"/*.yaml |
   awk -v threshold=${bicleaner_threshold} '{if ($3>threshold) {print $0}}' |
-  pigz >"${tmp_dir}/best.gz"
+  pigz >"${output_prefix}.best.gz"
 
 echo "### Writing output corpus"
-pigz -dc "${tmp_dir}/best.gz" | cut -f1 | pigz >"${output_prefix}.${SRC}.gz"
-pigz -dc "${tmp_dir}/best.gz" | cut -f2 | pigz >"${output_prefix}.${TRG}.gz"
+pigz -dc "${output_prefix}.best.gz" | cut -f1 | pigz >"${output_prefix}.${SRC}.gz"
+pigz -dc "${output_prefix}.best.gz" | cut -f2 | pigz >"${output_prefix}.${TRG}.gz"
```

Comment:
This can be accomplished in half the time up by using `tee`:
```
pigz -dc "${output_prefix}.best.gz" \
| tee >(cut -f1 | pigz >"${output_prefix}.${SRC}.gz") \
| cut -f2 | pigz >"${output_prefix}.${TRG}.gz"
```

Reply:
fixed